### ISSUE TITLE
User file fails with NAN in thermostat error message - may be due to room air model

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#5533]
## Issue overview

User file fails with this fatal error:

```
   ** Severe  ** DualSetPointWithDeadBand: Unanticipated combination of heating and cooling loads - report to EnergyPlus Development Team
   **   ~~~   ** occurs in Zone=2:CLASS1 During Warmup & Sizing, Environment=SUMMER DESIGN DAY IN JGSM JUL, at Simulation time=07/15 00:00 - 00:01
   **   ~~~   ** LoadToHeatingSetPoint=NAN, LoadToCoolingSetPoint=NAN
   **   ~~~   ** Zone Heating Set-point=21.11
   **   ~~~   ** Zone Cooling Set-point=24.44
   **   ~~~   ** Zone TempDepZnLd=22290.35
   **   ~~~   ** Zone TempIndZnLd=NAN
   **   ~~~   ** Zone ThermostatSetPoint=0.00
   **  Fatal  ** Program terminates due to above conditions.
```

There are warnings about the room air model inputs which may be a contributing factor, here's one example, but these appear for many zones:

```
   ** Warning ** RoomAirManager: Inconsistent ceiling heights in Zone: 2P:PLENUM
   **   ~~~   ** Lowest height=[8.534].
   **   ~~~   ** Highest height=[9.549].
   **   ~~~   ** Ceiling height=[3.611].
```
### Details

Some additional details for this issue (if relevant):
- Win64 on Windows 10
- 8.5.0 ffaabe11cd (develop as of 07 Mar 2016) and earlier versions
- Helpdesk ticket 11131
### Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [x] Defect file added to [DefectFiles](https://github.com/NREL/EnergyPlusDevSupport/tree/master/DefectFiles/5000s/5533)
- [x] Ticket added to Pivotal for defect (development team task)
- [x] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by lgu1234** [comment]
@mjwitte @Myoldmopar @EnergyArchmage I found a cause for NAN. It happens to a door surface: 2:Class1_Floor_9_0_8_0_8_0_Subsurface in the flowing code in the HcUCSDDV function of DisplacementVentMgr. The parent surface of the door surface is a floor: 2:Class1_Floor_9_0_8.

```
        // DOOR Hc, HA and HAT CALCULATION
        for ( Ctd = PosZ_Door( ( ZoneNum - 1 ) * 2 + 1 ); Ctd <= PosZ_Door( ( ZoneNum - 1 ) * 2 + 2 ); ++Ctd ) { // DOOR
            SurfNum = APos_Door( Ctd );
            Surface( SurfNum ).TAirRef = AdjacentAirTemp;
            if ( SurfNum == 0 ) continue;
            Z1 = minval( Surface( SurfNum ).Vertex( {1,Surface( SurfNum ).Sides} ), &Vector::z );
            Z2 = maxval( Surface( SurfNum ).Vertex( {1,Surface( SurfNum ).Sides} ), &Vector::z );
            ZSupSurf = Z2 - ZoneCeilingHeight( ( ZoneNum - 1 ) * 2 + 1 );
            ZInfSurf = Z1 - ZoneCeilingHeight( ( ZoneNum - 1 ) * 2 + 1 );

            if ( ZInfSurf > LayH ) {
                TempEffBulkAir( SurfNum ) = ZTMX( ZoneNum );
                CalcDetailedHcInForDVModel( SurfNum, TempSurfIn, DVHcIn );
                HDoor( Ctd ) = DVHcIn( SurfNum );
                HAT_MX += Surface( SurfNum ).Area * TempSurfIn( SurfNum ) * HDoor( Ctd );
                HA_MX += Surface( SurfNum ).Area * HDoor( Ctd );
            }

            if ( ZSupSurf < LayH ) {
                TempEffBulkAir( SurfNum ) = ZTOC( ZoneNum );
                CalcDetailedHcInForDVModel( SurfNum, TempSurfIn, DVHcIn );
                HDoor( Ctd ) = DVHcIn( SurfNum );
                HAT_OC += Surface( SurfNum ).Area * TempSurfIn( SurfNum ) * HDoor( Ctd );
                HA_OC += Surface( SurfNum ).Area * HDoor( Ctd );
            }

            if ( ZInfSurf <= LayH && ZSupSurf >= LayH ) {
                TempEffBulkAir( SurfNum ) = ZTMX( ZoneNum );
                CalcDetailedHcInForDVModel( SurfNum, TempSurfIn, DVHcIn );
                HLU = DVHcIn( SurfNum );
                TempEffBulkAir( SurfNum ) = ZTOC( ZoneNum );
                CalcDetailedHcInForDVModel( SurfNum, TempSurfIn, DVHcIn );
                HLD = DVHcIn( SurfNum );
                TmedDV = ( ( ZSupSurf - LayH ) * ZTMX( ZoneNum ) + ( LayH - ZInfSurf ) * ZTOC( ZoneNum ) ) / ( ZSupSurf - ZInfSurf );
                HDoor( Ctd ) = ( ( LayH - ZInfSurf ) * HLD + ( ZSupSurf - LayH ) * HLU ) / ( ZSupSurf - ZInfSurf );
                HAT_MX += Surface( SurfNum ).Area * ( ZSupSurf - LayH ) / ( ZSupSurf - ZInfSurf ) * TempSurfIn( SurfNum ) * HLU;
                HA_MX += Surface( SurfNum ).Area * ( ZSupSurf - LayH ) / ( ZSupSurf - ZInfSurf ) * HLU;
                HAT_OC += Surface( SurfNum ).Area * ( LayH - ZInfSurf ) / ( ZSupSurf - ZInfSurf ) * TempSurfIn( SurfNum ) * HLD;
                HA_OC += Surface( SurfNum ).Area * ( LayH - ZInfSurf ) / ( ZSupSurf - ZInfSurf ) * HLD;
                TempEffBulkAir( SurfNum ) = TmedDV;
            }

            DVHcIn( SurfNum ) = HDoor( Ctd );

        } // END DOOR
```

When (ZInfSurf == 0.0 && ZSupSurf == 0.0 && LayH == 0.0), the divide by zero occurs. If I treat this surface as a floor inside door section, the program run to completion.

```
            if ( ZInfSurf == 0.0 && ZSupSurf == 0.0 && LayH == 0.0 ) {
                TempEffBulkAir( SurfNum ) = ZTFloor( ZoneNum );
                CalcDetailedHcInForDVModel( SurfNum, TempSurfIn, DVHcIn );
                HFloor( Ctd ) = DVHcIn( SurfNum );
                HAT_FLOOR += Surface( SurfNum ).Area * TempSurfIn( SurfNum ) * HFloor( Ctd );
                HA_FLOOR += Surface( SurfNum ).Area * HFloor( Ctd );
                TempEffBulkAir( SurfNum ) = ZTFloor( ZoneNum );
                DVHcIn( SurfNum ) = HFloor( Ctd );
                continue;
            }
```

Here is my suggestion. We may treat a door surface as its parent. Please let me know what you think.

**Comment by mjwitte** [comment]
@lgu1234 I'm not sure I follow this completely, but treating the same as the base surface seems reasonable.

**Comment by lgu1234** [comment]
@mjwitte @Myoldmopar @EnergyArchmage I found a way to handle a door surface without base surface information. The window section in the same function treats a window surface with 3 subsections based on tilt: wall, ceiling, and floor. The existing Door section is used to handle wall only. After I mimic the window section in the Door section by adding ceiling and floor subsections, the problem went away and the program ran to completion.

By the way, there is a mistake in the input file:

   *\* Warning *\* InitOAController: Minimum Outdoor Air Flow Rate for Controller:OutdoorAir=AHU 1 AHU  OUTDOOR AIR CONTROLLER is greater than Design Supply Air Flow Rate for AirLoopHVAC=AHU 1.
   *\*   ~~~   *\* ...Minimum Outdoor Air Flow Rate=7.551159 will be reset to loop Design Supply Air Flow Rate=6.135317
   *\*  Fatal  *\* Error in Controller:OutdoorAir; program terminated

After I changed the Design Supply Air Flow Rate from 6.135 to 7.55, the warning went away. 

I will make a commit this morning.
